test/src/test/java/jenkins/model/NodesTest.java

```diff
@@ -88,6 +90,28 @@ public void addNodeShouldReplaceExistingNode() throws Exception {
         assertThat(r.jenkins.getNode("foo"), sameInstance(newNode));
     }
 
+    @Test
+    @Issue("JENKINS-56403")
+    public void replaceNodeShouldRemoveOldNode() throws Exception {
+        Node oldNode = r.createSlave("foo", "", null);
+        Node newNode = r.createSlave("foo-new", "", null);
+        r.jenkins.addNode(oldNode);
+        r.jenkins.getNodesObject().replaceNode(oldNode, newNode);
+        r.jenkins.getNodesObject().load();
+        assertNull(r.jenkins.getNode(oldNode.getNodeName()));
```

Comment:
```suggestion
        assertNull(r.jenkins.getNode("foo"));
```

since it is not otherwise obvious that the node’s name did not change somewhere in here.